### PR TITLE
CODEOWNERS: Remove file

### PR DESCRIPTION
**Pull request by russellb**
I propose dropping the CODEOWNERS file.

The use of CODEOWNERS prevents the default notifications
setting of "Participating and @mentioned" from working. If you are in
the CODEOWNERS, you get added to every PR and will get all
notifications accordingly. This is pretty painful on a busy repo. If a
maintainer desires that level of email notifications, they can change
their setting to "All Activity" for the repository.

CODEOWNERS is also not acting as access control to the repository.
That access is managed via github teams, and those teams are granted
different levels of access to the respository.

In summary, by not using a top-level entry like this, I feel we give
people more control over their notifications while not losing any
level of access controls on the repo. It allows people to stick to a
workflow that best works for them.

Signed-off-by: Russell Bryant <rbryant@redhat.com>

### CODE REVIEW

**Review by bjhargrave — APPROVED** [review]
LGTM. I am all for removing this file.

**Review by lehors — APPROVED** [review]
+1 The notifications are out of control! :-/
